### PR TITLE
[ko] cs-229-supervised-learning

### PR DESCRIPTION
**Pull request by kimdanny**
I've just made a new PR to work on top of rickie's translation.

### CODE REVIEW

**Comment by shervinea** [comment]
Thank you @kimdanny for building on top of @rickiepark's work! Marking the pull request as work in progress.

**Comment by kimdanny** [comment]
Just PR'ed the rest of the translations! @shervinea

**Comment by shervinea** [comment]
Thanks @kimdanny for all your work! Switching the PR to review mode. Please feel free to invite anyone you believe could take a look at your translation.

File: ko/cs-229-supervised-learning.md [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+
+**33. Generalized Linear Models**
+
+&#10230; 일반적 선형 모델 
+
+<br>
+
+**34. Exponential family ― A class of distributions is said to be in the exponential family if it can be written in terms of a natural parameter, also called the canonical parameter or link function, η, a sufficient statistic T(y) and a log-partition function a(η) as follows:**
+
+&#10230; 지수족 (Exponential family) - 어떠한 분포가 canonical paramter 또는 link function 이라고도 불리는 자연 모수 (natural parameter)에 관하여 정의될 수 있다면 그 분포는 지수족 안에 포함된다고 말할 수 있으며, canonical parameter (η), sufficient statistic (T(y)) 그리고 log-partition function (a(η)) 을 사용하여 다음과 같이 정의된다:
```
Comment (changh95) [inline ko/cs-229-supervised-learning.md:201]: spotted a type 😸 

```suggestion
&#10230; 지수족 (Exponential family) - 어떠한 분포가 canonical parameter 또는 link function 이라고도 불리는 자연 모수 (natural parameter)에 관하여 정의될 수 있다면 그 분포는 지수족 안에 포함된다고 말할 수 있으며, canonical parameter (η), sufficient statistic (T(y)) 그리고 log-partition function (a(η)) 을 사용하여 다음과 같이 정의된다: 
```

**Review by changh95 — APPROVED** [review]

File: ko/cs-229-supervised-learning.md (hunk 1 above)
Comment (kimdanny) [reply ko/cs-229-supervised-learning.md:201]: ouch! Thanks for spotting :)

File: ko/cs-229-supervised-learning.md [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+
+**49. In practice, the kernel K defined by K(x,z)=exp(−||x−z||22σ2) is called the Gaussian kernel and is commonly used.**
+
+&#10230; 보통은 커널 K는 K(x,z)=exp(−||x−z||22σ2)로 정의되며 이를 가우시안 커널이라고 한다. 
+
+<br>
+
+**50. [Non-linear separability, Use of a kernel mapping, Decision boundary in the original space]**
+
+&#10230; [비선형 구분가능, 커널 매핑의 사용, 원래의 공간에서의 결정경계]
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:297]: It is known that '에서의' is not good writing.
'원래의 공간에서의' --> '원본 공간에 있는'

File: ko/cs-229-supervised-learning.md [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+
+**50. [Non-linear separability, Use of a kernel mapping, Decision boundary in the original space]**
+
+&#10230; [비선형 구분가능, 커널 매핑의 사용, 원래의 공간에서의 결정경계] 
+
+<br>
+
+**51. Remark: we say that we use the "kernel trick" to compute the cost function using the kernel because we actually don't need to know the explicit mapping ϕ, which is often very complicated. Instead, only the values K(x,z) are needed.**
+
+&#10230; 비고: 우리는 보통 매우 복잡한 명시적 매핑 ϕ를 알 필요가 없기 때문에 커널을 사용하여 비용 함수를 계산하기 할 때 "커널 트릭"을 사용한다고 말한다. 이때 K(x,z) 값만 알면 된다.
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:303]: remove '우리는'
'계산하기 할 때' --> '계산할 때'

File: ko/cs-229-supervised-learning.md [hunk 4]
```diff
@@ -0,0 +1,567 @@
...
+
+**22. Linear regression**
+
+&#10230; 선형 회귀 
+
+<br>
+
+**23. We assume here that y|x;θ∼N(μ,σ2)**
+
+&#10230; 이때 우리는 y|x;θ∼N(μ,σ2) 라고 가정한다.
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:135]: Usually, 'you' or 'we' is not translated.
just remove '우리는'

File: ko/cs-229-supervised-learning.md [hunk 5]
```diff
@@ -0,0 +1,567 @@
...
+
+**29. Sigmoid function ― The sigmoid function g, also known as the logistic function, is defined as follows:**
+
+&#10230; 시그모이드 함수 - 로지스틱 함수 라고도 불리는 시그모이드 함수 g는, 다음과 같이 정의된다:
+
+<br>
+
+**30. Logistic regression ― We assume here that y|x;θ∼Bernoulli(ϕ). We have the following form:**
+
+&#10230; 로지스틱 회귀 - y|x;θ∼Bernoulli(ϕ) 라고 가정할 때 우리는 다음과 같은 식을 얻는다:
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:177]: remove '우리는'

File: ko/cs-229-supervised-learning.md [hunk 6]
```diff
@@ -0,0 +1,567 @@
...
+
+**54. Generative Learning**
+
+&#10230; 생성 학습
+
+<br>
+
+**55. A generative model first tries to learn how the data is generated by estimating P(x|y), which we can then use to estimate P(y|x) by using Bayes' rule.**
+
+&#10230; 생성 모델은 우선 데이터가 어떻게 생성되는지 P(x|y)를 추정하며 배우는데 우리는 이를 베이즈 정리를 사용하여 P(y|x)를 추정하는 데에 사용할 수 있다.
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:327]: remove '우리는'

File: ko/cs-229-supervised-learning.md [hunk 7]
```diff
@@ -0,0 +1,567 @@
...
+
+**59. Naive Bayes**
+
+&#10230; 나이브 베이즈
+
+<br>
+
+**60. Assumption ― The Naive Bayes model supposes that the features of each data point are all independent:**
+
+&#10230; 가정 - 나이브 베이즈 모델은 피처의 각각의 데이터들이 모두 서로 독립적이라고 가정한다:
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:357]: '피처의 각각의 데이터들이' --> '각 데이터 포인트의 특성이'

File: ko/cs-229-supervised-learning.md [hunk 8]
```diff
@@ -0,0 +1,567 @@
...
+
+**65. CART ― Classification and Regression Trees (CART), commonly known as decision trees, can be represented as binary trees. They have the advantage to be very interpretable.**
+
+&#10230; CART - 분류와 회귀트리(Classification and Regression Trees)의 줄임말로서 결정트리로 익히 알려져 있으며 이진트리로 표현될 수 있다. 이 모델은 해석이 가능하다는 장점이 있다.
+
+<br>
+
+**66. Random forest ― It is a tree-based technique that uses a high number of decision trees built out of randomly selected sets of features. Contrary to the simple decision tree, it is highly uninterpretable but its generally good performance makes it a popular algorithm.**
+
+&#10230; 랜덤 포레스트 - 무작위로 선택된 피처들로부터 형성된 많은 수의 의사 결정 트리들을 사용하는 트리 기반 기술이다. 단순한 의사 결정 트리와는 달리 해석이 매우 어렵지만 일반적으로 좋은 성능을 내어 널리 사용되는 알고리즘이다.
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:393]: '피처들' --> '특성으'

File: ko/cs-229-supervised-learning.md [hunk 9]
```diff
@@ -0,0 +1,567 @@
...
+
+**67. Remark: random forests are a type of ensemble methods.**
+
+&#10230; 비고: 랜덤 포레스트는 앙상블 방법 중의 하나이다.
+
+<br>
+
+**68. Boosting ― The idea of boosting methods is to combine several weak learners to form a stronger one. The main ones are summed up in the table below:**
+
+&#10230; 부스팅 - 부스팅 방법은 여러 약한 학습기(weak learners)들을 합쳐서 더 강한 학습기를 만드는 것이다. 주요 목록들이 밑 테이블에 있다:
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:405]: For easy reading, '밑' --> '아래'

File: ko/cs-229-supervised-learning.md [hunk 10]
```diff
@@ -0,0 +1,567 @@
...
+
+<br>
+
+**69. [Adaptive boosting, Gradient boosting]**
+
+&#10230; [아다 부스팅(Adaptive boosting), 그라디언트 부스팅(Gradient boosting)]
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:411]: '아다' --> '에이다'

File: ko/cs-229-supervised-learning.md [hunk 11]
```diff
@@ -0,0 +1,567 @@
...
+
+<br>
+
+**70. High weights are put on errors to improve at the next boosting step**
+
+&#10230; 다음 부스팅 스텝에서 더 나아지기 위해 에러에 높은 가중치가 매겨지게 된다
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:417]: '더 나아지기 위해 에러에 높은 가중치가 매겨지게 된다' --> '성능을 높이기 위해 에러 샘플에 높은 가중치를 부여한다'

File: ko/cs-229-supervised-learning.md [hunk 12]
```diff
@@ -0,0 +1,567 @@
...
+
+<br>
+
+**71. Weak learners trained on remaining errors**
+
+&#10230; 남아있는 에러로 학습된 약한 학습기
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:423]: '남아있는 에러로' --> '잔차에서'

File: ko/cs-229-supervised-learning.md [hunk 13]
```diff
@@ -0,0 +1,567 @@
...
+
+**71. Weak learners trained on remaining errors**
+
+&#10230; 남아있는 에러로 학습된 약한 학습기 
+
+<br>
+
+**72. Other non-parametric approaches**
+
+&#10230; 다른 non-parametric 접근법들
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:429]: Usually, plural is not translated.
non-parametric 접근법들 --> 비모수(non-parametric) 접근법

File: ko/cs-229-supervised-learning.md [hunk 14]
```diff
@@ -0,0 +1,567 @@
...
+
+**82. the training examples are drawn independently**
+
+&#10230; 학습 데이터들은 모두 독립적으로 추출 되었다
+
+<br>
+
+**83. Shattering ― Given a set S={x(1),...,x(d)}, and a set of classifiers H, we say that H shatters S if for any set of labels {y(1),...,y(d)}, we have:**
+
+&#10230; Shattering - 집합 S={x(1),...,x(d)}와 분류기의 집합 H가 주어졌을 때, 아래와 같은 조건을 만족한다면 레이블 {y(1),...,y(d)}에 대해 H가 S를 shatter 한다고 한다:
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:495]: Shattering --> 쉐터링(Shattering)
shatter 한다고 --> 쉐터링한다고

File: ko/cs-229-supervised-learning.md [hunk 15]
```diff
@@ -0,0 +1,567 @@
...
+
+**83. Shattering ― Given a set S={x(1),...,x(d)}, and a set of classifiers H, we say that H shatters S if for any set of labels {y(1),...,y(d)}, we have:**
+
+&#10230; Shattering - 집합 S={x(1),...,x(d)}와 분류기의 집합 H가 주어졌을 때, 아래와 같은 조건을 만족한다면 레이블 {y(1),...,y(d)}에 대해 H가 S를 shatter 한다고 한다: 
+
+<br>
+
+**84. Upper bound theorem ― Let H be a finite hypothesis class such that |H|=k and let δ and the sample size m be fixed. Then, with probability of at least 1−δ, we have:**
+
+&#10230; Upper bound 이론 - H를 |H|=k를 만족하는 유한 가설 클래스로 정하고 δ와 표본 크기 m을 고정한다. 그러면 적어도 1−δ의 확률로 다음을 얻는다:
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:501]: Upper bound 이론 --> 상계 이론(Upper bound theorem)

File: ko/cs-229-supervised-learning.md [hunk 16]
```diff
@@ -0,0 +1,567 @@
...
+
+**84. Upper bound theorem ― Let H be a finite hypothesis class such that |H|=k and let δ and the sample size m be fixed. Then, with probability of at least 1−δ, we have:**
+
+&#10230; Upper bound 이론 - H를 |H|=k를 만족하는 유한 가설 클래스로 정하고 δ와 표본 크기 m을 고정한다. 그러면 적어도 1−δ의 확률로 다음을 얻는다: 
+
+<br>
+
+**85. VC dimension ― The Vapnik-Chervonenkis (VC) dimension of a given infinite hypothesis class H, noted VC(H) is the size of the largest set that is shattered by H.**
+
+&#10230; VC 차원 ― 주어진 무한 가설 클래스 H의 Vapnik-Chervonenkis (VC) 차원, VC(H)는 H에 의해 shatter 되는 가장 큰 집합의 크기이다.
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:507]: Vapnik-Chervonenkis (VC) 차원 --> VC(Vapnik-Chervonenkis) 차원
shatter 되는 --> 쉐터링되는

File: ko/cs-229-supervised-learning.md [hunk 17]
```diff
@@ -0,0 +1,567 @@
...
+
+**89. [Notations and general concepts, loss function, gradient descent, likelihood]**
+
+&#10230; [표기법과 일반적 개념, 손실 함수, 경사 하강법, 가능도]
+
+<br>
+
+**90. [Linear models, linear regression, logistic regression, generalized linear models]**
+
+&#10230; [선형 모델, 선형 회귀, 로지스틱 회귀, 일반화되 선형 모델]
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:537]: 일반화되 --> 일반화된

File: ko/cs-229-supervised-learning.md [hunk 18]
```diff
@@ -0,0 +1,567 @@
...
+
+**38. Assumptions of GLMs ― Generalized Linear Models (GLM) aim at predicting a random variable y as a function fo x∈Rn+1 and rely on the following 3 assumptions:**
+
+&#10230; GLM의 가정 - 일반화 선형 모델 (GLM)은 확률변수(random variable) y를 x∈Rn+1 의 함수로서 예측하는 것에 목표를 두며 다음 3가지 가정에 의존한다: 
+
+<br>
+
+**39. Remark: ordinary least squares and logistic regression are special cases of generalized linear models.**
+
+&#10230; 최소자승법 (ordinary least squares)과 로지스틱 회귀는 일반화 선형 모델의 특수한 경우이다.
```
Comment (rickiepark) [inline ko/cs-229-supervised-learning.md:231]: 일반화 --> 일반화된

**Review by rickiepark — COMMENTED** [review]
Hi @kimdanny ,
Thank you for this pr.
Please check my suggestions. :)

**Comment by kimdanny** [comment]
Thanks to @rickiepark 's wonderful and very detailed review, I could fix some of my mistakes of mine. His review definitely shows his professionalism. 
I believe that this work has now reached a launch-able quality @shervinea :)

**Comment by rickiepark** [comment]
Thank you @kimdanny 
LGTM! :)

**Comment by shervinea** [comment]
Thank you both @rickiepark and @kimdanny for your amazing work!

**Pull request merged**